logstash/config/logstash.conf

```diff
@@ -22,13 +22,14 @@ filter {
     }
   }
 
-  #Grokking Spring Boot's default log format
+  #Grokking Spring Boot's default log format + Spring Sleuth tracing data
   grok {
-    match => [ "message",
-               "(?<timestamp>%{YEAR}-%{MONTHNUM}-%{MONTHDAY} %{TIME})\s*%{LOGLEVEL:level} %{NUMBER:pid} --- \[(?<thread>[A-Za-z0-9-]+)\] [A-Za-z0-9.]*\.(?<class>[A-Za-z0-9#_]+)\s*:\s+(?<logmessage>.*)",
-               "message",
-               "(?<timestamp>%{YEAR}-%{MONTHNUM}-%{MONTHDAY} %{TIME})\s*%{LOGLEVEL:level} %{NUMBER:pid} --- .+? :\s+(?<logmessage>.*)"
-             ]
+      match => [ "message", "(?<timestamp>%{YEAR}-%{MONTHNUM}-%{MONTHDAY} %{TIME}) *%{LOGLEVEL:level} \[%{DATA:application},%{DATA:traceId},%{DATA:spanId},%{DATA:zipkin}] %{DATA:pid} --- *\[ *%{DATA:thread}] %{JAVACLASS:class} *: %{GREEDYDATA:logmessage}",
+                 "message", "(?<timestamp>%{YEAR}-%{MONTHNUM}-%{MONTHDAY} %{TIME}) *%{LOGLEVEL:level} \[%{DATA:application},%{DATA:traceId},%{DATA:spanId},%{DATA:zipkin}] %{DATA:pid} --- *\[ *%{DATA:thread}] .*: %{GREEDYDATA:logmessage}",
+                 "message", "(?<timestamp>%{YEAR}-%{MONTHNUM}-%{MONTHDAY} %{TIME}) *%{LOGLEVEL:level} %{NUMBER:pid} --- *\[(?<thread>[A-Za-z0-9-]+)\] [A-Za-z0-9.]*\.(?<class>[A-Za-z0-9#_]+)\s*: %{GREEDYDATA:logmessage}",
+                 "message", "(?<timestamp>%{YEAR}-%{MONTHNUM}-%{MONTHDAY} %{TIME}) *%{LOGLEVEL:level} %{NUMBER:pid} --- .+? : %{GREEDYDATA:logmessage}"
+               ]
+
```

Comment:
In many cases you are using "<space>*". Isn't it better to use "\s*", the whitespace shortcut for matching spaces as well as tabs.